src/node.rs

```diff
@@ -130,6 +130,29 @@ impl<T> Node<T> {
     }
 
     ///
+    /// Exchange the attached data through a new one
```

Comment:
What about "Replaces this `Node`s data with the data provided".
The swap is explained in the next line and this also sounds wrong